Complete all tasks for Lab#1

bodkiny opened the pull request:
## Created the required classes and implemented them according to the task description. Covered all classes with unit tests with 100% line coverage.

![image](https://github.com/bodkiny/nure-cross-platform-programming/assets/124298580/ff2218dd-a23d-4305-96ce-9dbe2dd6fba3)

### Libraries used:
- 'org.junit.jupiter:junit-jupiter-api:5.10.2';
- 'org.junit.jupiter:junit-jupiter-engine:5.10.2';
- 'org.projectlombok:lombok:1.18.30'.

### Plugins:
- 'org.apache.maven.plugins:maven-surefire-plugin:3.2.5';
- 'org.jacoco:jacoco-maven-plugin:0.8.11'.

bodkiny committed:
Implement StringFromBinaryTree and add tests

This commit implements the StringFromBinaryTree class according to the task description. It includes a method to build a string representation of a binary tree. Corresponding tests have also been added to verify the correct functionality of the implementation.

bodkiny committed:
Implement SortingNumbers class and add tests

This commit implements the SortingNumbers class, which includes a method to sort an array of integers using the insertion sort algorithm. The class also includes a method to handle user input and validate it.

In addition, this commit includes the SortingNumbersTest class, which contains unit tests for the SortingNumbers class. These tests verify the correct functionality of the sorting algorithm and the user input handling.

bodkiny committed:
Add Lombok configuration and dependency

Configure Lombok to not include generated methods in JaCoCo coverage reports

bodkiny committed:
Implement ComplexNumber and Point classes with corresponding tests

kdimak reviewed (changes requested):
First impression: really good! I left small remarks from what the most important one is to split `SortingNumbers` class into two.

bodkiny committed:
Add JUnit dependency for parametrized tests

bodkiny committed:
Refactor: Decompose SortingNumbers and SortingNumbersTest classes to adhere to SOLID principles

- Removed SortingNumbers and SortingNumbersTest classes.
- Introduced new classes and interfaces: TaskProcessor, TaskProcessorTest, ConsoleArrayReader, InputStreamArrayReader, InsertionSorter, Sorter, and ArrayReader.
- TaskProcessor class is responsible for processing tasks which depends on the Sorter and ArrayReader interfaces adhering the Liskov Substitution Principle.
- TaskProcessorTest class contains unit tests for TaskProcessor.
- ConsoleArrayReader and InputStreamArrayReader classes are responsible for reading arrays.
- InsertionSorter class implements the Sorter interface and is responsible for sorting arrays.
- The Sorter and ArrayReader interfaces were introduced to adhere to the Dependency Inversion and Interface Segregation Principles.
- The changes ensure that each class has a single responsibility, adhering to the Single Responsibility Principle.
- The system is now more flexible and easier to maintain, as changes to low-level modules will not affect high-level modules.

bodkiny committed:
Enhancement: Add non-static add and multiply methods to ComplexNumber class

- Added non-static add and multiply methods to ComplexNumber class.
- These methods operate on the current instance of ComplexNumber.
- The static add and multiply methods are still present and functional.
- Updated the ComplexNumberTest class to include test cases for the new non-static add and multiply methods.

bodkiny committed:
Refactor: Remove disabled test

- Removed disabled test due to method implementation

bodkiny committed:
Refactor: Reorganize classes into new packages for better code readability

- Moved `Sorter` and `InsertionSorter` classes to the new package `ua.nure.cpp.lab1.computations.sorters`.
- Moved `ArrayReader`, `ConsoleArrayReader`, and `InputStreamArrayReader` classes to the new package `ua.nure.cpp.lab1.computations.readers`.

bodkiny committed:
Update: Add GitHub Actions workflow for running tests & generating coverage report

Added a GitHub Actions workflow file named 'workflow.yml' to automate running tests and generating a code coverage report for the project. The workflow is triggered on pushes to the 'dev' branch and pull requests targeting 'master' or 'dev' branches.

The workflow includes two jobs:
  1. 'build-run-tests': Sets up JDK 17, runs Maven verify phase, and copies JaCoCo HTML report to generate coverage report.
  2. 'upload-coverage-report': Uploads the generated coverage report artifacts to GitHub Action page.

bodkiny committed:
Update: Add GitHub Actions workflow for running tests & generating coverage report

Added a GitHub Actions workflow file named 'workflow.yml' to automate running tests and generating a code coverage report for the project. The workflow is triggered on pushes to the 'dev' branch and pull requests targeting 'master' or 'dev' branches.

The workflow includes two jobs:
  1. 'build-run-tests': Sets up JDK 17, runs Maven verify phase, and copies JaCoCo HTML report to generate coverage report.
  2. 'upload-coverage-report': Uploads the generated coverage report artifacts to GitHub Action page.

bodkiny committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	.github/workflows/build-test-report.yml

bodkiny commented on lab1/src/test/java/ua/nure/cpp/lab1/computations/TaskProcessorTest.java:
```diff
@@ -0,0 +1,59 @@
+    @MethodSource("provideTestCases")
+    void testSorting(String input, int[] expected) {
+        InputStream in = new ByteArrayInputStream(input.getBytes());
+        System.setIn(in);
```
I forgot to remove it. It stayed after I attempted to initialize the `System.in` input stream in the `ConsoleArrayReader` constructor without parameters. But in that case, the `System.setIn()` method wasn't working for some reason.
